django_auth_adfs/token_manager.py

```diff
@@ -0,0 +1,489 @@
+        if not hasattr(request, "session"):
+            return False
+
+        try:
+            session_modified = False
+
+            encrypted_token = self.encrypt_token(access_token)
+            if encrypted_token:
+                request.session[self.ACCESS_TOKEN_KEY] = encrypted_token
+                session_modified = True
+                logger.debug("Stored access token")
+
+            if adfs_response and "refresh_token" in adfs_response:
+                refresh_token = adfs_response["refresh_token"]
+                if refresh_token:
+                    encrypted_token = self.encrypt_token(refresh_token)
+                    if encrypted_token:
+                        request.session[self.REFRESH_TOKEN_KEY] = encrypted_token
+                        session_modified = True
+                        logger.debug("Stored refresh token")
+                    else:
+                        logger.warning("Failed to encrypt refresh token")
+                else:
+                    logger.warning("Empty refresh token received from ADFS")
+            else:
+                logger.debug("No refresh token in ADFS response")
+
+            if adfs_response and "expires_in" in adfs_response:
+                expires_at = datetime.datetime.now() + datetime.timedelta(
+                    seconds=int(adfs_response["expires_in"])
+                )
+                request.session[self.TOKEN_EXPIRES_AT_KEY] = expires_at.isoformat()
+                session_modified = True
+                logger.debug("Stored token expiration")
+
+            if self.store_obo_token:
+                try:
+                    from django_auth_adfs.backend import AdfsBaseBackend
+
+                    backend = AdfsBaseBackend()
+                    obo_token = backend.get_obo_access_token(access_token)
+                    if obo_token:
+                        encrypted_token = self.encrypt_token(obo_token)
+                        if encrypted_token:
+                            request.session[self.OBO_ACCESS_TOKEN_KEY] = encrypted_token
+                            import jwt
+
+                            decoded_token = jwt.decode(
+                                obo_token, options={"verify_signature": False}
+                            )
+                            if "exp" in decoded_token:
+                                obo_expires_at = datetime.datetime.fromtimestamp(
+                                    decoded_token["exp"]
+                                )
+                                request.session[self.OBO_TOKEN_EXPIRES_AT_KEY] = (
+                                    obo_expires_at.isoformat()
+                                )
+                                session_modified = True
+                                logger.debug(
+                                    "Stored OBO token with expiration from token claims"
+                                )
+                except Exception as e:
```

Comment:
There are several blank Exception catches. That's typically an anti-pattern. Is there a good reason to employ it in this file?